.circleci/config.yml

```diff
@@ -27,28 +27,52 @@ experiment_job: &experiment_job
     EXP: << parameters.exp >>
 
 executors:
-  amphtml-small-executor:
+  docker-small:
     docker:
       - image: cimg/base:stable
     resource_class: small
-  amphtml-medium-executor:
+  docker-medium:
     docker:
       - image: cimg/node:lts-browsers
     resource_class: medium
-  amphtml-large-executor:
+  docker-large:
     docker:
       - image: cimg/node:lts-browsers
     resource_class: large
-  amphtml-xlarge-executor:
+  docker-xlarge:
     docker:
       - image: cimg/node:lts-browsers
     resource_class: xlarge
-  validator-xlarge-executor:
+  jdk-docker-xlarge:
     docker:
       - image: cimg/openjdk:16.0-node
     resource_class: xlarge
+  macos-medium:
+    macos:
+      xcode: 12.4.0
+    resource_class: medium
 
 commands:
+  checkout_repo:
+    steps:
+      - restore_cache:
+          name: 'Restore Git Cache'
+          keys:
+            - git-cache-{{ arch }}-{{ .Branch }}-{{ .Revision }}
+            - git-cache-{{ arch }}-{{ .Branch }}-
+            - git-cache-{{ arch }}-
+      - checkout
+      - save_cache:
+          name: 'Save Git Cache'
+          key: git-cache-{{ arch }}-{{ .Branch }}-{{ .Revision }}
+          paths:
+            - .git
+  setup_node_environment:
+    steps:
+      - node/install:
+          lts: true
+          install-npm: false
```

Comment:
Except for the macos machine, why do we need to install node? Node is already installed, by definition, on the `node` docker instances...

Reply:
I did this for three reasons:

1. In theory, CircleCI node VMs are supposed to have the most recent LTS version of nodejs, but in practice, this isn't guaranteed until we update this line in our config (which regularly goes out of date):
https://github.com/ampproject/amphtml/blob/3170de88a344b50e5b903df739381b7423e5a5eb/.circleci/config.yml#L5
2. On docker VMs, there used to be a need to overwrite the default node prefix directory because the system default location couldn't be written to. Installing LTS node using `nvm` makes that step unnecessary.
https://github.com/ampproject/amphtml/blob/3170de88a344b50e5b903df739381b7423e5a5eb/.circleci/config.yml#L78-L79
3. The time overhead for installing LTS node is very small (4-5 seconds), and seems like a good price to pay to guarantee that there's no unexpected variation in the node version across VMs.

Happy to revisit this if can think of any downsides that I haven't considered.